woocommerce/woocommerce
# Skip the need to click the Connect button when the connection is initiated from WCCOM.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR redirects the user to the connection confirmation screen when initiating the connection from WooCommerce.com, when a URL parameter is present. Currently, the user is redirected to the `My Subscriptions` tab under `WooCommerce > Extensions` page in their store's admin area, where they must click the connect Connect button.

WooCommerce will start attaching the `connect` URL param when a connection is started from the `My Stores` page on WooCommerce.com ([PR 22462](https://github.com/Automattic/woocommerce.com/pull/22462)). If the URL parameter is present, and after proper checks finish, the user is redirected to the connection approval screen on WooCommerce.com instead of loading the `My subscriptions` page in their store, where they wold have to click connect again.

This is backward compatible and previous versions of WooCommerce will continue behaving as they did before. They'll ignore the URL param. 

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. In your local woocommerce.test environment checkout the `add/22378-streamline-wccom-connection` branch. This is the matching branch on WCCOM that has the necessary changes to test this end-to-end.
2. Checkout this branch in your `woocommerce` local setup.
3. Run `npm run build:zip` from `<woocommerce repo root>/plugins/woocommerce`
4. Upload the build `<woocommerce repo root>/plugins/woocommerce/woocommerce.zip` to your [customer test site](https://github.com/Automattic/woocommerce.com/blob/trunk/docs/customer-test-site.md). 
5. Navigate to https://woocommerce.test/my-account/my-stores/ and click the `Connect store` button, input your test store URL (where you uploaded the built zip) and click connect.
6. Confirm that you are taken straight to the confirm connection screen on woocommerce.test and clicking the `connect your store` establishes the connection and takes you back to the My Stores page on woocommerce.test, where you can see the store listed.
7. Confirm edge cases are covered:
    - Clicking cancel on the confirmation page cancels the connection.
    - You can disconnect the store the same as before.
    - No weird redirects happen when navigating the store's admin area (WooCommerce > Extensions)
8. Please test around this issue.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
